core/src/main/java/com/linecorp/armeria/common/RequestContextAwareCompletableFuture.java

```diff
@@ -243,4 +253,96 @@
     public CompletableFuture<T> exceptionally(Function<Throwable, ? extends T> fn) {
         return ctx.makeContextAware(super.exceptionally(ctx.makeContextAware(fn)));
     }
+
+    // support JDK9 functions
+
+    public <U> CompletableFuture<U> newIncompleteFuture() {
+        return ctx.makeContextAware(new CompletableFuture<>());
+    }
+
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<T> completeAsync(Supplier<? extends T> supplier) {
+        final Callable<? extends T> callable = Functions.toCallable(supplier);
+        return (CompletableFuture<T>) ctx.makeContextAware(supplyAsync(
+                Functions.fromCallable(ctx.makeContextAware(callable))));
+    }
+
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<T> completeAsync(Supplier<? extends T> supplier,
+                                              Executor executor) {
+        final Callable<? extends T> callable = Functions.toCallable(supplier);
+        return (CompletableFuture<T>) ctx.makeContextAware(supplyAsync
+                (Functions.fromCallable(ctx.makeContextAware(callable)), executor));
+    }
+
+    public CompletableFuture<T> orTimeout(long timeout, TimeUnit unit) {
+        if (unit == null)
+            throw new NullPointerException();
+        if (!this.isDone())
+            whenComplete(new Canceller(Delayer.delay(new Timeout(this),
+                    timeout, unit)));
+        return ctx.makeContextAware(this);
+    }
+
+    public CompletableFuture<T> completeOnTimeout(T value, long timeout,
+                                                  TimeUnit unit) {
+        if (unit == null)
+            throw new NullPointerException();
+        if (!this.isDone())
+            whenComplete(new Canceller(Delayer.delay(
+                    new DelayedCompleter<>(this, value),
+                    timeout, unit)));
+        return ctx.makeContextAware(this);
+    }
+
+    static final class Delayer {
+        static ScheduledFuture<?> delay(Runnable command, long delay,
+                                        TimeUnit unit) {
+            return delayer.schedule(command, delay, unit);
+        }
+
+        static final class DaemonThreadFactory implements ThreadFactory {
+            public Thread newThread(Runnable r) {
+                Thread t = new Thread(r);
+                t.setDaemon(true);
+                t.setName("CompletableFutureDelayScheduler");
```

Comment:
How about prepending `armeria-`, not to be confused with the thread created by JDK?